garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,262 @@ local meta = FindMetaTable( "Vector" )
+-----------------------------------------------------------]]
+function meta:Resize(num)
+	self:Normalize()
+	self:Mul(num)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the resized vector
+-----------------------------------------------------------]]
+function meta:GetResized(...)
+	local v = Vector(self)
+	v:Resize(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to resized midpoint of itself
+-----------------------------------------------------------]]
+function meta:Midpoint(vec)
+	self:Add(vec)
+	self:Mul(0.5)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the midpoint vector
+-----------------------------------------------------------]]
+function meta:GetMidpoint(...)
+	local v = Vector(self)
+	v:Midpoint(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the zero point vector
+-----------------------------------------------------------]]
+function meta:GetZero(...)
+	local v = Vector(self)
+	v:Zero(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Applies new basis to a vector by the three axises of an angle
+-----------------------------------------------------------]]
+function meta:Basis(ang)
+	local x = self:Dot(ang:Forward())
+	local y = self:Dot(ang:Right())
+	local z = self:Dot(ang:Up())
+	self:SetUnpacked(x, y, z)
+end
+
+--[[---------------------------------------------------------
+Applies new basis to a vector by the three axises of an angle
+-----------------------------------------------------------]]
+function meta:GetBasis(...)
+	local v = Vector(self)
+	v:Basis(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Projects the vector in to an axis defined by the second vector
+-----------------------------------------------------------]]
+function meta:Project(vec)
+	local vn = vec:GetNormalized()
+	local nm = self:Dot(vn)
+	self:Set(vn)
+	self:Mul(nm)
+end
+
+--[[---------------------------------------------------------
+Projects the vector in to an axis defined by the second vector
+-----------------------------------------------------------]]
+function meta:GetProject(...)
+	local v = Vector(self)
+	v:Project(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Retrieves the angle between two vectors from 0 to math.pi
+-----------------------------------------------------------]]
+function GetAngle(vec)
+	local x = self:GetProject(vec)
+	local y = self:GetSub(x)
+	local nx = x:Dot(vec)
+	local mx = nx/math.abs(nx)
+	return math.atan2(y:Length(), x:Length()*nx)
+end
+
+--[[---------------------------------------------------------
+Calculates 3x3 determinant by using other two vectors as rows
+-----------------------------------------------------------]]
+function meta:GetDeterminant(vec2, vec3)

```

Comment:
I would just make the name ``Determinant`` to be consistent with ``Vector:Length``.

Reply:
Ok. Fixed.